[ENHANCEMENT] Add support for context in migrations

Glenn-Ewing opened the pull request:
**Summary:**
Abandons previous implementation of context enabled migrations (which added context support at the handler and Migration levels) in favor of this much simpler approach, which allows context to be passed into the finder and extracts the values thereof into the symbol table at the time the migrations are read in, allowing for simple interpolation.

**Pros/Cons with Change**
Previous approach implemented twig templating for readability withExpectedContext() for testing/validation; so the Con is that this approach leaves migrations a bit more open to developer error, but the pros of the implementation being dead simple seem far outweigh these cons by far.

Thanks to @joshmcrae for the insight/simpler implementation!

Glenn-Ewing committed:
Update Finder.php

Enable context in Finder.

Glenn-Ewing committed:
Update Finder.php

Make extract usage conditional.

Glenn-Ewing reviewed (commented):
Self-review completed

joshmcrae reviewed (changes requested):
Great @Glenn-Ewing! Could we please add a test case to cover usage of the `Finder` class' new `$context` property.

There's already fixtures under `tests/Migrations` which should make it easy to test that variables are extracted.

Glenn-Ewing commented:
> Great @Glenn-Ewing! Could we please add a test case to cover usage of the `Finder` class' new `$context` property.
> 
> There's already fixtures under `tests/Migrations` which should make it easy to test that variables are extracted.

Derp, Sure thing, will do!

Glenn-Ewing committed:
Add Tests for Context Enabled Migrations

1. Adds a context enabled migration (with context default, where normally an error might be thrown).
2. Tests that the finder accepts and extracts context for use in migrations.
3. Tests sad path, ensureing that context is undefined as expected and can be coded for as desired within the migration file(s).

Glenn-Ewing commented:
Hey @joshmcrae ! I've added test to allow for passing context through the finder to be used in migrations, etc.

Glenn-Ewing reviewed (commented):
Self-review completed.

joshmcrae commented on src/Util/Finder.php:
```diff
@@ -4,8 +4,22 @@
 
-class Finder
-{
+class Finder {
```
The opening brace should be on the next line for code-style consistency.

Glenn-Ewing replied:
Right on, this is usually auto-handled for me, so I guess Ive gotten lazy. I would've thought code lint would catch that, but it may not be enabled on this repo yet?

Glenn-Ewing replied:
Done, see latest commit.

joshmcrae reviewed (changes requested):
Just a minor code-style fix and this is ready to go.

Glenn-Ewing committed:
Update Finder.php

Lint fix

Glenn-Ewing reviewed (commented):
Self-review completed.

joshmcrae commented:
Thanks @Glenn-Ewing!

Released as [1.2.0](https://github.com/tithely/exo/releases/tag/1.2.0)
